ext/opentelemetry-ext-zipkin/src/opentelemetry/ext/zipkin/__init__.py

```diff
@@ -0,0 +1,182 @@
+        ipv6: Optional[str] = None,
+    ):
+        self.service_name = service_name
+        self.host_name = host_name
+        self.port = port
+        self.endpoint = endpoint
+        self.protocol = protocol
+        self.url = "{}://{}:{}{}".format(
+            self.protocol, self.host_name, self.port, self.endpoint
+        )
+        self.ipv4 = ipv4
+        self.ipv6 = ipv6
+
+    def export(self, spans: Sequence[Span]) -> "SpanExportResult":
+        zipkin_spans = self._translate_to_zipkin(spans)
+        result = requests.post(
+            url=self.url, data=json.dumps(zipkin_spans), headers=ZIPKIN_HEADERS
+        )
+
+        if result.status_code not in SUCCESS_STATUS_CODE:
+            logger.error(
+                "Traces cannot be uploaded; status code: %s, message %s",
+                result.status_code,
+                result.text,
+            )
+            # TODO: should we retry here?
+            return SpanExportResult.FAILED_NOT_RETRYABLE
+        return SpanExportResult.SUCCESS
+
+    def _translate_to_zipkin(self, spans: Sequence[Span]):
+
+        local_endpoint = {
+            "serviceName": self.service_name,
+            "port": self.port,
+        }
+
+        if self.ipv4 is not None:
+            local_endpoint["ipv4"] = self.ipv4
+
+        if self.ipv6 is not None:
+            local_endpoint["ipv6"] = self.ipv6
+
+        zipkin_spans = []
+        for span in spans:
+            ctx = span.get_context()
+            trace_id = ctx.trace_id
+            span_id = ctx.span_id
+
+            # Timestamp in zipkin spans is int of microseconds.
+            start_timestamp_mus = _nsec_to_usec_round(span.start_time)
+            duration_mus = _nsec_to_usec_round(span.end_time - span.start_time)
+
+            zipkin_span = {
+                "traceId": _format(trace_id),
+                "id": _format(span_id),
+                "name": span.name,
+                "timestamp": start_timestamp_mus,
+                "duration": duration_mus,
+                "localEndpoint": local_endpoint,
+                "kind": SPAN_KIND_MAP[span.kind],
+                "tags": _extract_tags_from_span(span.attributes),
+                "annotations": _extract_annotations_from_events(span.events),
+            }
+
+            if isinstance(span.parent, Span):
+                zipkin_span["parentId"] = _format(
+                    span.parent.get_context().span_id
+                )
+            elif isinstance(span.parent, SpanContext):
+                zipkin_span["parentId"] = _format(span.parent.span_id)
+
+            zipkin_spans.append(zipkin_span)
+        return zipkin_spans
+
+    def shutdown(self) -> None:
+        pass
+
+
+def _format(unformatted_id):
+    return format(unformatted_id, "x")
+
+
+def _extract_tags_from_span(attr):
+    if not attr:
+        return None
+    tags = {}
+    for attribute_key, attribute_value in attr.items():
+        if isinstance(attribute_value, (int, bool, float)):
```

Comment:
is this case allowed? it seems like we should do some enforcement on the creation of attributes spans, rather than try to catch it on the flush side here.

Reply:
Attributes are assigned one of the following values: https://github.com/open-telemetry/opentelemetry-python/blob/531d0b166eb3102e0f4c38b30ceb69ecf66be1dc/opentelemetry-api/src/opentelemetry/util/types.py#L18

Looks like int is not one of them though, i'll remove it.

Reply:
Although that isn't enforced at this point... I can create a separate ticket for this

https://github.com/open-telemetry/opentelemetry-python/blob/baab508ff92b333cf7310ddd46c966808b825102/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L211

Reply:
I'm in issues anyway: https://github.com/open-telemetry/opentelemetry-python/issues/347. Thanks!